src/hal/drivers/mesa_pktgyro_test.comp

```diff
@@ -0,0 +1,288 @@
+instances to enable. A check is included to ensure that the count= option is
+not used instead.
+For simplicity we test only one PktUART instance, therefore load the component
+like this:
+
+\\fB loadrt mesa_uart names=hm2_5i25.0.pktuart.0\\fR
+
+The PktUART instance names are printed to the dmesg buffer during the Hostmot2
+setup sequence, one for each PktUART instance included in the bitfile loaded to
+each installed card during the Hostmot2 setup sequence. Type "dmesg" at the
+terminal prompt to view the output.
+If you want to work with more than one PktUART instance, consult Andy Pugh's
+mesa_uart.comp
+
+In order to compile and install do:
+\\fB halcompile --install src/hal/drivers/mesa_pktgyro_test.comp\\fR
+
+The component exports only one function, namely receive, which needs to be added
+to a realtime thread.
+To test this component  set DEBUG=5 before and execute this HAL script:
+\\fB loadrt hostmot2\\fR
+\\fB loadrt hm2_pci\\fR
+\\fB loadrt mesa_pktgyro_test names=hm2_5i25.0.pktuart.0\\fR
+\\fB loadrt threads name1=test1 period1=10000000\\fR
+\\fB addf hm2_5i25.0.pktuart.0.receive test1\\fR
+\\fB start\\fR
+
+Check linuxcnc.log for debug output.
+
+""";
+
+
+author "Boris Skegin";
+license "GPL";
+
+include <unistd.h>;
+include "../drivers/mesa-hostmot2/hostmot2.h";
+
+
+pin out s32 rxbytes  "Number of Bytes received or negative Error code";
+
+variable char *name; // PktUART name
+
+option extra_setup yes;
+
+function receive;
+
+;;
+
+
+
+#define BAUDRATE (115200)
+
+
+
+/* This uses the RTAPI_MP_ARRAY_STRING macro to load the list of PktUART channels
+into an array. This is copied into the *name string of each */
+
+char *pktuart_chans[4] = {0,};
+RTAPI_MP_ARRAY_STRING(pktuart_chans, 2, "PktUART Channel names");
+
+static hostmot2_t* hm2=NULL;
+
+
+
+FUNCTION(receive){
+	rtapi_u16 max_frame_length = 20;
+	rtapi_u8 num_frames = 20;
+	unsigned char Replyd3[num_frames*max_frame_length];
+	rtapi_u16 frame_sizes3[20];
+	rxbytes=hm2_pktuart_read(name, Replyd3, &num_frames, &max_frame_length, frame_sizes3);
+	rtapi_print_msg(RTAPI_MSG_INFO, "PktUART receive: got %d bytes, %d frames\n", rxbytes, num_frames);
+
+	//Print out the actual frame sizes
+	int i;
+	for(i=0;i<num_frames; i++) {
+	    rtapi_print_msg(RTAPI_MSG_INFO, "Rec frame %d: size %d bytes\n", i+1 , frame_sizes3[i]);
+		}
+	rtapi_print_msg(RTAPI_MSG_INFO, "\n");
+
+	// Print out ACK replies for the frames sent out
+	// from EXTRA_SETUP function
+	int k=0;
+	int bytes_total = 0;
+	while(k <num_frames) {
+		        int j=0;
+
+	            while(j<=frame_sizes3[k]-1) {
+				   rtapi_print_msg(RTAPI_MSG_INFO, "Rec frame %d, byte %d 0x%.2x", k+1 , bytes_total+j , Replyd3[bytes_total+j]);
+				   j++;
+				}
+		   bytes_total = bytes_total + frame_sizes3[k];
+		   rtapi_print_msg(RTAPI_MSG_INFO, "\n");
+		   k++;
+	}
+
+
+
+}
+
+EXTRA_SETUP(){ // the names parameters are passed in 'prefix'. You just have to know that.
```

Comment:
Yes, that line and the lines just following.

I think I understand that the comment is complaining that the docs don't mention that 'prefix' is an implicit argument of EXTRA_SETUP.  If so, a better option is to improve the documentation instead of scattering comments that the documentation is incomplete!

I don't understand why the first character of the instance prefix should be forbidden to be "m".  Can you tell me more about that?  It sounds like it's a heavy-handed way to prevent use of count=, that ends up preventing legitimate use as well.